fundingmanager.go

```diff
@@ -837,23 +849,23 @@ func (f *fundingManager) handleFundingOpen(fmsg *fundingOpenMsg) {
 	openChan, err := resCtx.reservation.FinalizeReservation()
 	if err != nil {
 		fndgLog.Errorf("unable to finalize reservation: %v", err)
-		fmsg.peer.Disconnect()
 		return
 	}
 
 	// The reservation has been completed, therefore we can stop tracking
 	// it within our active reservations map.
-	f.deleteReservationCtx(peerID, chanID)
+	f.deleteReservationCtx(peerKey, chanID)
 
-	fndgLog.Infof("FundingOpen: ChannelPoint(%v) with peerID(%v) is now open",
-		resCtx.reservation.FundingOutpoint(), peerID)
+	fndgLog.Infof("FundingOpen: ChannelPoint(%v) with peerKey(%v) is now open",
+		resCtx.reservation.FundingOutpoint(), peerKey)
 
 	// Notify the L3 routing manager of the newly active channel link.
 	// TODO(roasbeef): should have sigs, only after funding_locked is
 	// recv'd
 	//  * also ensure fault tolerance, scan opened chan on start up check
 	//  for graph existence
-	f.announceChannel(fmsg.peer.server, openChan, fmsg.msg.ChanChainID,
+	peer := f.findPeer(peerKey)
+	f.announceChannel(peer.server, openChan, fmsg.msg.ChanChainID,
```

Comment:
I think the `ChannelRouter` can be passed directly into this method rather than a pointer to the server.